packages/sil-governance/src/routes/dashboard/components/Proposal/CollapsingData/DisplayOptions/index.tsx

```diff
@@ -0,0 +1,43 @@
+import {
+  ActionKind,
+  CreateTextResolutionAction,
+  ProposalAction,
+  ReleaseEscrowAction,
+  SendAction,
+  SetValidatorsAction,
+  UpdateElectionRuleAction,
+  UpdateElectorateAction,
+} from "@iov/bns";
+import React from "react";
+
+import CreateTextResolution from "./CreateTextResolution";
+import ReleaseEscrow from "./ReleaseEscrow";
+import Send from "./Send";
+import SetValidators from "./SetValidators";
+import UpdateElectionRule from "./UpdateElectionRule";
+import UpdateElectorate from "./UpdateElectorate";
+
+interface Props {
+  readonly action: ProposalAction;
+}
+
+const DisplayOptions = ({ action }: Props): JSX.Element => {
+  switch (action.kind) {
+    case ActionKind.CreateTextResolution:
+      return <CreateTextResolution action={action as CreateTextResolutionAction} />;
```

Comment:
Hm, it's pretty annoying that we need these casts.

Reply:
Looks like we don't need then :)

<img width="982" alt="Bildschirmfoto 2019-08-29 um 23 08 45" src="https://user-images.githubusercontent.com/2603011/63976632-08ac2680-cab2-11e9-82c8-098cfe6935e1.png">